src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -299,7 +300,7 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
       breadcrumbs,
       discoveredBreadcrumbs,
       stickyNotifications: resolvedStickyNotifications,
-      navigationOpen,
+      navigationOpen: !!resolvedNavigation && navigationOpen,
```

Comment:
resolvedNavigation value of null will ensure the navigation is not opened when navigationHidden.